library/waku_thread/waku_thread.nim

```diff
@@ -0,0 +1,188 @@
+
+{.pragma: exported, exportc, cdecl, raises: [].}
+{.pragma: callback, cdecl, raises: [], gcsafe.}
+{.passc: "-fPIC".}
+
+import
+  std/[json,sequtils,times,strformat,options,atomics,strutils,os]
+import
+  chronicles,
+  chronos,
+  stew/results,
+  stew/shims/net
+import
+  ../../../waku/common/enr/builder,
+  ../../../waku/v2/waku_enr/capabilities,
+  ../../../waku/v2/waku_enr/multiaddr,
+  ../../../waku/v2/waku_enr/sharding,
+  ../../../waku/v2/waku_core/message/message,
+  ../../../waku/v2/waku_core/topics/pubsub_topic,
+  ../../../waku/v2/node/peer_manager/peer_manager,
+  ../../../waku/v2/waku_core,
+  ../../../waku/v2/node/waku_node,
+  ../../../waku/v2/node/builder,
+  ../../../waku/v2/node/config,
+  ../../../waku/v2/waku_relay/protocol,
+  ../events/[json_error_event,json_message_event,json_base_event],
+  ../alloc,
+  ./config,
+  ./inter_thread_communication/request
+
+type
+  Context* = object
+    thread: Thread[(ptr Context)]
+    reqChannel: Channel[InterThreadRequest]
```

Comment:
`Channel` uses some memory allocator tricks that are not necessarily safe - it has been removed in v2 / moved outside of the standard library because it's design has some problems - in general, it's likely better to use a serialization library that creates a flat representation of the data, allocates a shared memory section and passes that to the other thread - the easiest one to start with is probably json-serialization - later a more efficient one can be selected.

https://gist.github.com/arnetheduck/b6a7ac8f4b85490d26d464674e09d57d is a simple example of this approach.

In general, `ref` types are allocated on a thread-local heap and using them in cross-thread scenarios is not well-tested in nim - it's safer to stick with non-ref types completely and rely on "manual" allocation with allocShared / createShared for this purpose.

We will eventually develop a common high-performance channel, but that doesn't really matter for a use case like waku.